tests/e2e/mod.rs

```diff
@@ -0,0 +1,118 @@
+use std::collections::HashMap;
+use std::process::Command;
+use std::time::Duration;
+
+use anyhow::Error;
+use camino::Utf8PathBuf;
+use duct::cmd;
+use git2::Repository;
+use reqwest::Client;
+use rstest::*;
+use serde::{Deserialize, Serialize};
+use serde_json::json;
+use tempfile::TempDir;
+use tokio::time::timeout;
+
+mod dev;
+
+const GRAPHQL_TIMEOUT_DURATION: Duration = Duration::from_secs(30);
+
+#[derive(Debug, Deserialize)]
+struct ReducedSuperGraphConfig {
+    subgraphs: HashMap<String, ReducedSubgraphConfig>,
+}
+#[derive(Debug, Deserialize)]
+struct ReducedSubgraphConfig {
+    routing_url: String,
+}
+
+impl ReducedSuperGraphConfig {
+    pub fn get_subgraph_urls(self) -> Vec<String> {
+        self.subgraphs
+            .values()
+            .map(|x| x.routing_url.clone())
+            .collect()
+    }
+}
+
+#[fixture]
+#[once]
+fn run_subgraphs_retail_supergraph() -> TempDir {
+    println!("Cloning required git repository");
+    // Clone the Git Repository that's needed to a temporary folder
+    let cloned_dir = TempDir::new().expect("Could not create temporary directory");
+    Repository::clone(
+        "https://github.com/apollosolutions/retail-supergraph",
+        cloned_dir.path(),
+    )
+    .expect("Could not clone supergraph repository");
+    // Jump into that temporary folder and run npm commands to kick off subgraphs
+    println!("Installing subgraph dependencies");
+    cmd!("npm", "install")
+        .dir(cloned_dir.path())
+        .run()
+        .expect("Could not install subgraph dependencies");
+    cmd!("npm", "install", "-g", "nodemon")
+        .dir(cloned_dir.path())
+        .run()
+        .expect("Could not install nodemon");
+    println!("Kicking off subgraphs");
+    let mut cmd = Command::new("npx");
+    cmd.env("NODE_ENV", "dev");
+    cmd.args(["nodemon", "index.js"]).current_dir(&cloned_dir);
+    cmd.spawn().expect("Could not spawn subgraph process");
+    println!("Finding subgraph URLs");
+    let subgraph_urls = get_subgraph_urls(
+        Utf8PathBuf::from_path_buf(cloned_dir.path().join("supergraph-config-dev.yaml"))
+            .expect("Could not create path to config"),
+    );
+    println!("Testing subgraph connectivity");
+    for subgraph_url in subgraph_urls {
+        tokio::task::block_in_place(|| {
+            let client = Client::new();
+            let handle = tokio::runtime::Handle::current();
+            handle.block_on(async {
+                timeout(
+                    GRAPHQL_TIMEOUT_DURATION,
+                    test_graphql_connection(&client, &subgraph_url),
+                )
+                .await
+                .expect("Exceeded maximum time allowed")
+            })
+        })
+        .expect("Could not execute connectivity check");
+    }
```

Comment:
I appreciate this isn't the most efficient if you have a very large number of subgraphs, I tried mapping the `subgraph_urls` into a Vec of Future but couldn't make it all hang together. Again if there's a better way I'm all ears!

Reply:
Is it worth us picking a random subset instead? How many URLs do we envisage here? Is it likely to grow?

Reply:
This is only once for tests, so I'm not too concerned about the performance just yet 👀

Reply:
Also it's pulling from a specific example at which the rate of change is pretty glacial, so I'm not enormously worried